src/redash_stmo/datasource_link/web/index.js

```diff
@@ -0,0 +1,14 @@
+import template from './datasource-link.html';
+
+function controller($scope, $route) {
+  const dataSources = $route.current.locals.dataSources;
+  const query = $route.current.locals.query;
+  $scope.dataSource = dataSources.find(ds => ds.id === query.data_source_id);
```

Comment:
I understand that we use Angular's resource thing here to get the data from the API, and I'm not sure how to write the actual API call to the new datasource link API endpoint. But since we'll be doing React in the future, might as well not lean too much on Angular IMO.

In fact, don't you want to try writing this in React instead? Allen could help you get up to speed how to do that and the React setup is already in Redash enabled.